### PR TITLE
i#1312 AVX-512 support: Fix AVX vcvtss2sd, vcvtsd2ss's destination in decode table.

### PR DESCRIPTION
**Pull request by hgreving2304**
Submission 738842b missed to fix some of the convert opcodes.

Issue: #1312

### CODE REVIEW

**Comment by hgreving2304** [comment]
I inspected all the other ss/sd opcodes again, didn't find any other bug. I just looked again and yet found more bugs in existing AVX opcodes. Hold off with review, I will update this PR.

**Comment by hgreving2304** [comment]
I found that all the existing f(n)madd and f(n)msub ss and sd avx opcodes are also wrong. Fixing this requires removing the create macros that provide API to take the destination as implicit source for these opcodes.

**Comment by hgreving2304** [comment]
PTAL.

File: core/arch/x86/decode_table.c [hunk 1]
```diff
@@ -6279,14 +6279,14 @@ const instr_info_t vex_W_extensions[][2] = {
     {OP_vfmadd231ps,0x6638b818,"vfmadd231ps",Vvs,xx,Hvs,Wvs,Vvs,mrm|vex|reqp,x,END_LIST},
     {OP_vfmadd231pd,0x6638b858,"vfmadd231pd",Vvs,xx,Hvs,Wvs,Vvs,mrm|vex|reqp,x,END_LIST},
   }, { /* vex_W_ext 3 */
-    {OP_vfmadd132ss,0x66389918,"vfmadd132ss",Vss,xx,Hss,Wss,Vss,mrm|vex|reqp,x,END_LIST},
-    {OP_vfmadd132sd,0x66389958,"vfmadd132sd",Vsd,xx,Hsd,Wsd,Vsd,mrm|vex|reqp,x,END_LIST},
+    {OP_vfmadd132ss,0x66389918,"vfmadd132ss",Vdq,xx,Hss,Wss,Vdq,mrm|vex|reqp,x,END_LIST},
```
Comment (derekbruening) [inline core/arch/x86/decode_table.c]: The manual shows this instruction touching only the bottom 32 bits of the destination in all encoding forms, and zeroing bits above 128: bits 32..127 are untouched.  That does not sound like a "dq" size.

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: Which manual are you looking at? Sometimes they add former erratas.
https://software.intel.com/sites/default/files/managed/39/c5/325462-sdm-vol-1-2abcd-3abcd.pdf
5-138 Vol. 2C
States DEST[127:32] <- DEST[127:32] for VEX and EVEX for all forms.

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: Exactly: `DEST[127:32] <- DEST[127:32]`, i.e., a nop, left untouched.

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: Ok :) That's true. I would argue this is semantically still the entire register, you could argue it stems from the fact that DEST is an implicit source. It is definitely not Vss. Should we make up a new type that describes the "untouched" part here, or "say" Vdq? I would choose Vdq. DR's attempt to abstract the operand's size is not always precise. What about masked writes for example in AVX-512? WDYT

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: No, we cannot consider this to write the whole register.  If we consider this to write the whole register, we'd have to consider "mov $0,%al" to write the entire %eax.  I don't see any relation to the dest being a source.

A taint tracker has to know that this is not touching all the bits.  We can't over-estimate what's touched.

This looks exactly like Vss to me: OPSZ_4_of_16.  Touching just the bottom 4 bytes.

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: It's definitely not Vss. EVEX as well as VEX do DEST[MAX_VL-1:128] <- 0.

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: I guess it would be Vss for xmm, but something else for ymm and zmm.

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: That always happens for writes to xmm by vex-encoded (or evex-encoded) instructions: the top half of ymm is zeroed.

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: I don't think that's modeled in the IR at all and tools have to deal with that themselves.  We provide instr_zeroes_ymmh() for this purpose.

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: We consider it similar to how for 64-bit x86, if you write to %eax, it zeroes the top half of %rax: we don't explicitly encode that inside the IR sizes.

Comment (derekbruening) [reply core/arch/x86/decode_table.c]: This instruction only operates on xmm right?

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: Well! If that's how it is currently defined, I will revert all the vfma* in this PR!

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: > This instruction only operates on xmm right?

Yes. That said, Vss does make more sense then. I'll revert this part.

Comment (hgreving2304) [reply core/arch/x86/decode_table.c]: Done.

**Pull request merged**